internal/charmstore/store_test.go

```diff
@@ -1432,3 +1432,125 @@ func (s *StoreSuite) TestAddBundleArchiveIndexed(c *gc.C) {
 	c.Assert(err, gc.IsNil)
 	s.checkAddBundle(c, bundleArchive, true)
 }
+
+var getPromulgatedURLTests = []struct {
+	about    string
```

Comment:
It would be nice to have a test with more than one previously promulgated charms, so that we check the last revision is used to calculate the new one.